packages/mui-joy/src/Tooltip/Tooltip.tsx

```diff
@@ -0,0 +1,789 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import clsx from 'clsx';
+import {
+  appendOwnerState,
+  PopperUnstyled,
+  unstable_composeClasses as composeClasses,
+} from '@mui/base';
+import { MUIStyledCommonProps } from '@mui/system';
+import { OverridableComponent } from '@mui/types';
+import {
+  unstable_capitalize as capitalize,
+  unstable_useControlled as useControlled,
+  unstable_useEventCallback as useEventCallback,
+  unstable_useForkRef as useForkRef,
+  unstable_useIsFocusVisible as useIsFocusVisible,
+} from '@mui/utils';
+import styled from '../styles/styled';
+import { Theme } from '../styles/types';
+import useThemeProps from '../styles/useThemeProps';
+import { getTooltipUtilityClass } from './tooltipClasses';
+import { TooltipOwnerState, TooltipProps, TooltipTypeMap } from './TooltipProps';
+
+const useUtilityClasses = (ownerState: TooltipProps) => {
+  const { disableInteractive, arrow, placement, variant, color, size } = ownerState;
+
+  const slots = {
+    root: [
+      'root',
+      placement && `placement${capitalize(placement.split('-')[0])}`,
+      size && `size${capitalize(size)}`,
+      color && `color${capitalize(color)}`,
+      variant && `variant${capitalize(variant)}`,
+      !disableInteractive && 'rootInteractive',
+      arrow && 'rootArrow',
```

Comment:
```suggestion
      arrow && 'hasArrow',
```
Is this more explicit?